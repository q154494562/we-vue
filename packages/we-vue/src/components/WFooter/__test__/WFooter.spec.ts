import { mount, shallowMount } from '@vue/test-utils'
import Footer from '../'

describe('footer', () => {
  test('create', () => {
    const wrapper = shallowMount(Footer, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-footer')
    expect(wrapper.classes()).toContain('weui-footer')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('text', () => {
    const wrapper = shallowMount(Footer, {
      propsData: {
        text: 'test',
      },
    })

    expect(wrapper.contains('p.weui-footer__text')).toBeTruthy()
    expect(wrapper.find('p.weui-footer__text').text()).toBe('test')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render with links', () => {
    const wrapper = mount(Footer, {
      propsData: {
        text: 'test',
        links: [
          {
            text: 'foo',
            link: 'bar',
          },
          {
            text: 'foo',
            link: 'bar',
          },
        ],
      },
    })

    expect(wrapper.findAll('.weui-footer__link')).toHaveLength(2)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
