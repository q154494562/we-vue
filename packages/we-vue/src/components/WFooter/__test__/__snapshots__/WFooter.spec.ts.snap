// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer create 1`] = `
<div class="weui-footer">
  <p class="weui-footer__text"></p>
</div>
`;

exports[`footer render with links 1`] = `
<div class="weui-footer">
  <p class="weui-footer__links">
    <div class="weui-footer__link">foo</div>
    <div class="weui-footer__link">foo</div>
  </p>
  <p class="weui-footer__text">test</p>
</div>
`;

exports[`footer text 1`] = `
<div class="weui-footer">
  <p class="weui-footer__text">test</p>
</div>
`;
